cronjob_scripts/trigger-package-build.py

```diff
@@ -0,0 +1,217 @@
+from cronjob_scripts.checkout_worktree import checkout_worktree_for_arch
+from cronjob_scripts.stats import get_requested_crates
+
+TARGET_ARCH_TO_BUILD_OS = {
+    "x86_64-apple-darwin": "macos-latest",
+    "aarch64-apple-darwin": "macos-latest",
+    "x86_64-unknown-linux-gnu": "ubuntu-20.04",
+    "x86_64-unknown-linux-musl": "ubuntu-20.04",
+    "x86_64-pc-windows-msvc": "windows-latest",
+    "aarch64-pc-windows-msvc": "windows-latest",
+    "aarch64-unknown-linux-gnu": "ubuntu-20.04",
+    "aarch64-unknown-linux-musl": "ubuntu-20.04",
+    "armv7-unknown-linux-musleabihf": "ubuntu-20.04",
+    "armv7-unknown-linux-gnueabihf": "ubuntu-20.04",
+}
+
+
+def main():
+    target_arches = get_target_architectures()
+    for target_arch in target_arches:
+        trigger_for_arch(target_arch)
+
+
+def trigger_for_arch(target_arch: str):
+    print(f"Triggering builds for {target_arch}")
+
+    build_os = get_build_os(target_arch)
+
+    tracking_worktree_path = checkout_worktree_for_arch(target_arch)
+    excludes = get_excludes(tracking_worktree_path, days=7, max_failures=5)
+
+    popular_crates = get_popular_crates()
+    requested_crates = get_requested_crates(period="1 day", arch=target_arch)
+
+    possible_crates = set(popular_crates + requested_crates) - set(excludes)
+
+    # check random crates up to a limit.
+    check_limit = int(os.environ.get("CRATE_CHECK_LIMIT", 5))
+    # Note that the old next-unbuilt-package.sh script actually checked the top 4 * CRATE_CHECK_LIMIT
+    # crates from get-stats.sh + popular-crates.txt rather than shuffling the whole lot before checking.
+    # The original script just looped through the whole list, but kept track of where it got to,
+    # to avoid getting stuck on unbuildable crates.
+    #
+    # For now I'm trying to be roughly backwards compatible with the old script's numbers,
+    # but I think that shuffling first is an important bugfixt. Once we are using crates.io sparse
+    # index api without rate-limits, we can think about checking more crates.
+    crates_to_check = random.sample(list(possible_crates), 4 * check_limit)
+    if os.environ.get("RECHECK"):
+        crates_to_check = ["cargo-quickinstall"] + crates_to_check
+
+    repo_url = get_repo_url()
+    branch = get_branch()
+    builds_scheduled = 0
+    for crate in crates_to_check:
+        print(f"Checking {crate} for {target_arch}")
+        version = get_current_version_if_unbuilt(
+            repo_url=repo_url, crate=crate, target_arch=target_arch
+        )
+        if not version:
+            continue
+
+        no_default_features = "false"
+        if crate == "gitoxide":
+            features = "max-pure"
+            no_default_features = "true"
+        elif crate == "sccache":
+            features = "vendored-openssl"
+        else:
+            features = ",".join(
+                [feat for feat in version["features"].keys() if "vendored" in feat]
```

Comment:
Nit: could use a generator expression:

```suggestion
                feat for feat in version["features"].keys() if "vendored" in feat
```